src/ticket.c

```diff
@@ -1159,42 +1199,55 @@ void set_ticket_wakeup(struct ticket_config *tk)
 {
 	timetype near_future, tv, next_vote;
 
+	set_future_time(&near_future, 10);
+
 	/* At least every hour, perhaps sooner (default) */
+	tk_log_debug("ticket will be woken up after up to one hour");
 	ticket_next_cron_in(tk, 3600*TIME_RES);
-	set_future_time(&near_future, 10);
 
 	switch (tk->state) {
 	case ST_LEADER:
-		assert(tk->leader == local);
-
-		get_next_election_time(tk, &next_vote);
-
-		/* If timestamp is in the past, wakeup in
-		 * near future */
-		if (!is_time_set(&next_vote)) {
-			tk_log_debug("next ts unset, wakeup soon");
-			ticket_next_cron_at(tk, &near_future);
-		} else if (is_past(&next_vote)) {
-			int tdiff = time_left(&next_vote);
-			tk_log_debug("next ts in the past " intfmt(tdiff));
-			ticket_next_cron_at(tk, &near_future);
-		} else {
-			ticket_next_cron_at(tk, &next_vote);
+		if (tk->mode != TICKET_MODE_MANUAL) {
+			/* Leader won't initialize new elections for manual tickets */
+			assert(tk->leader == local);
+
+			get_next_election_time(tk, &next_vote);
+
+			/* If timestamp is in the past, wakeup in
+			 * near future */
+			if (!is_time_set(&next_vote)) {
+				tk_log_debug("next ts unset, wakeup soon");
+				ticket_next_cron_at(tk, &near_future);
+			} else if (is_past(&next_vote)) {
+				int tdiff = time_left(&next_vote);
+				tk_log_debug("next ts in the past " intfmt(tdiff));
+				ticket_next_cron_at(tk, &near_future);
+			} else {
+				ticket_next_cron_at(tk, &next_vote);
+			}
 		}
 		break;
 
 	case ST_CANDIDATE:
-		assert(is_time_set(&tk->election_end));
-		ticket_next_cron_at(tk, &tk->election_end);
+		if (tk->mode != TICKET_MODE_MANUAL) {
+			/* There is no Candidate state for manual tickets */
+			assert(is_time_set(&tk->election_end));
+			ticket_next_cron_at(tk, &tk->election_end);
+		}
 		break;
 
 	case ST_INIT:
 	case ST_FOLLOWER:
 		/* If there is (or should be) some owner, check on it later on.
 		 * If no one is interested - don't care. */
 		if (is_owned(tk)) {
-			interval_add(&tk->term_expires, tk->acquire_after, &tv);
-			ticket_next_cron_at(tk, &tv);
+			if (tk->mode != TICKET_MODE_MANUAL) {
+				interval_add(&tk->term_expires, tk->acquire_after, &tv);
+				ticket_next_cron_at(tk, &tv);
+			} else {
+				/* Set next cron some half minute after acquire_after time */
+				ticket_next_cron_in(tk, 30*TIME_RES + tk->acquire_after);
```

Comment:
There being no failover for manual tickets, is this needed at all? Shouldn't manual tickets just be idle, i.e. all further processing obsoleted?

Reply:
Agreed. As a follower, we don't expect any actions, so we shouldn't plan any cron tasks here.